lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -1520,7 +1520,11 @@ namespace Js
         ScriptContext* scriptContext = instance->GetScriptContext();
         DynamicType* cachedDynamicType = nullptr;
         BOOL isJsrtExternalType = instance->GetType()->IsJsrtExternal();
-        DynamicType* oldType = isJsrtExternalType ? 0 : instance->GetDynamicType();
+        // Share the type of all JsrtExternal objects which will have 'newPrototype' as their new prototype.
+        // TypeTransitionMap is a map with key as weakReference and to store it, key need to have a valid heap block.
+        // So instead of using some constant, use newPrototype as the key for saving newType for such objects
+        // in TypeTransitionMap.
+        DynamicType* oldType = isJsrtExternalType ? reinterpret_cast<DynamicType*>(newPrototype) : instance->GetDynamicType();
```

Comment:
Can you explain how you can share the same type for all JsrtExternal objects?  what if the current JsrtExternalObject has some properties already? should the newType with the newPrototype be necessary be different then one that is for JsrtExternalObject that has no property?